unit_tests/utils/test_team_repo.py

```diff
@@ -1,10 +1,95 @@
 # -*- coding: utf-8 -*-
+import os
+
 from elitefurretai.utils.team_repo import TeamRepo
 
 
 def test_team_repo():
-    tr = TeamRepo("data/fixture")
-    assert "example_team" in tr.teams
+    tr = TeamRepo("data/teams")
+    assert "gen9vgc2024regg_speed_ability1" in tr.teams
+    assert "gen8vgc2020_s6_yan" in tr.teams
     assert len(tr.teams) > 5
     for team in tr.teams:
         assert isinstance(tr.teams[team], str)
+
+
+# helper function
+def get_paste(filepath: str) -> str:
+    with open(filepath, "r") as f:
+        return f.read()
+
+
+def test_validator():
+    teams_folder = "data/teams"
+    tr = TeamRepo(teams_folder)
+    formats = os.listdir(teams_folder)
+
+    # test invalidity for all formats
+    blank = get_paste("data/fixture/test_teams/blank")
+    gibberish = get_paste("data/fixture/test_teams/gibberish.txt")
+    illegal_set = get_paste("data/fixture/test_teams/illegal_moveset.txt")
+    invalid_pastes = [blank, gibberish, illegal_set]
+
+    for format_name in formats:
+        for paste in invalid_pastes:
+            assert tr.validate_team(paste, format_name) is False
+
+    # test validity for all formats
+    format_name = "gen8vgc2020"
+    assert tr.validate_team(tr.teams["gen8vgc2020_s2_koutesh"], format_name) is True
```

Comment:
Small note -- assert's test if something is True, so the `is True` here is redundant